index.js

```diff
@@ -1,186 +1,259 @@
 'use strict';
 
-var FXOSSimulators = require('node-firefox-find-simulators');
-var Q = require('q');
+// See https://github.com/jshint/jshint/issues/1747 for context
+/* global -Promise */
+var Promise = require('es6-promise').Promise;
 var net = require('net');
-var FXPorts = require('node-firefox-ports');
 var spawn = require('child_process').spawn;
-var FirefoxClient = require('firefox-client');
-var portfinder = require('portfinder');
 var fs = require('fs');
-var __ = require('underscore');
-
-
-module.exports = startB2G;
-
-function portIsReady(port, cb) {
-  var defer = Q.defer();
-
-  function ping() {
-    var sock = new net.Socket();
-    sock
-      .on('connect', function() {
-        defer.resolve();
-        sock.destroy();
-      })
-      .on('error', function(e) {
-        if (e && e.code !== 'ECONNREFUSED') {
-          throw e;
-        }
-        sock.destroy();
-        setTimeout(function() {
-          ping(defer);
-        }, 1000);
-      })
-      .connect(port,'localhost');
-  }
-  ping();
-  return defer.promise;
-}
+var portFinder = require('portfinder');
+var findSimulators = require('node-firefox-find-simulators');
 
-function commandB2G(opts) {
-  var defer = Q.defer();
+module.exports = startSimulator;
 
-  var childOptions = { stdio: ['ignore', 'ignore', 'ignore'] };
+function startSimulator(options) {
 
-  if (opts.exit) {
-    childOptions.detached = true;
-  }
+  var detached = options.detached ? true : false;
+  var verbose = options.verbose ? true : false;
+  var port = options.port;
 
-  if (opts.verbose) {
-    childOptions.stdio = [process.stdin,  process.stdout, process.stderr];
+  var simulatorOptions = {};
+  if (options.version) {
+    simulatorOptions.version = options.version;
   }
 
-  if (opts.stdin) {
-    childOptions.stdio[0] = fs.openSync(opts.stdin, 'a');
-  }
-  if (opts.stdout) {
-    childOptions.stdio[1] = fs.openSync(opts.stdout, 'a');
-  }
-  if (opts.stderr) {
-    childOptions.stdio[2] = fs.openSync(opts.stderr, 'a');
-  }
+  return new Promise(function(resolve, reject) {
 
-  var simProcess = spawn(
-    opts.bin,
-    ['-profile', opts.profile, '-start-debugger-server', opts.port, '-no-remote'],
-    childOptions
-  );
+    Promise.all([ findSimulator(simulatorOptions), findAvailablePort(port) ])
+      .then(function(results) {
```

Comment:
The first `.then` being on a newline leads to it being much harder for me to match up the indentation of `Promise.all` with the code that runs after it. I keep expecting

```javascript
Promise.all().then(function() {
  // Some stuff.
});
```